types/datasource.d.ts

```diff
@@ -76,28 +86,194 @@ import {IsolationLevel, Transaction} from './transaction-mixin';
  */
 export declare class DataSource extends EventEmitter {
   name: string;
-  settings: Options;
+  settings: ConnectorSettings;
```

Comment:
Based on the codebase, the DataSource simply passes the whole settings to the connector; Hence why the type called `ConnectorSettings`.

Reply:
On second thought, we may want to create a dedicated passthrough type to avoid potential future issues due to the shared type.